### PR TITLE
fix roboelectric unit test

### LINKED ISSUE #4698: Robolectric unit testing example should be updated to 3.3.1+
According to https://github.com/robolectric/robolectric/issues/2944 (related to https://github.com/realm/realm-java/issues/3808#issuecomment-303660221 ), Robolectric yet again supports Powermock, with version 3.3.1+, so updating unit test sample would be nice.

### PR DESCRIPTION
**Pull request by abennsir**
-Upgrade to Robolectric 3.3.2 
closes #4698

### CODE REVIEW

File: examples/unitTestExample/src/test/java/io/realm/examples/unittesting/ExampleActivityTest.java [hunk 1]
```diff
@@ -64,14 +66,15 @@
...
+@PowerMockRunnerDelegate(RobolectricTestRunner.class)
 @Config(constants = BuildConfig.class, sdk = 21)
 @PowerMockIgnore({"org.mockito.*", "org.robolectric.*", "android.*"})
 @SuppressStaticInitializationFor("io.realm.internal.Util")
 @PrepareForTest({Realm.class, RealmConfiguration.class, RealmQuery.class, RealmResults.class, RealmCore.class, RealmLog.class})
-public class ExampleActivityTest {
-
-    // Robolectric, Using Power Mock https://github.com/robolectric/robolectric/wiki/Using-PowerMock
+public class ExampleActivityTest
+{
```
Comment (dalinaum) [inline examples/unitTestExample/src/test/java/io/realm/examples/unittesting/ExampleActivityTest.java]: Should not it be split. It should be after `ExampleActivityTest`.

File: examples/unitTestExample/src/test/java/io/realm/examples/unittesting/ExampleActivityTest.java [hunk 2]
```diff
@@ -64,14 +66,15 @@
...
+	// Robolectric, Using Power Mock https://github.com/robolectric/robolectric/wiki/Using-PowerMock
```
Comment (dalinaum) [inline examples/unitTestExample/src/test/java/io/realm/examples/unittesting/ExampleActivityTest.java]: Please use 4 spaces instead.

File: examples/unitTestExample/src/test/java/io/realm/examples/unittesting/ExampleActivityTest.java [hunk 3]
```diff
@@ -176,16 +179,14 @@ public void setup() throws Exception {
...
 
-    @Ignore("FIXME: Some problems mocking OKHttp")
     @Test
     public void shouldBeAbleToAccessActivityAndVerifyRealmInteractions() {
         doCallRealMethod().when(mockRealm).executeTransaction(Mockito.any(Realm.Transaction.class));
 
         // Create activity
-        ActivityController<ExampleActivity> controller = Robolectric.buildActivity(ExampleActivity.class).setup();
-        ExampleActivity activity = controller.get();
+	    ExampleActivity activity = Robolectric.buildActivity(ExampleActivity.class).create().start().resume().visible().get();
```
Comment (dalinaum) [inline examples/unitTestExample/src/test/java/io/realm/examples/unittesting/ExampleActivityTest.java]: Wrong indentation.

File: examples/unitTestExample/src/test/java/io/realm/examples/unittesting/ExampleActivityTest.java [hunk 4]
```diff
@@ -176,16 +179,14 @@ public void setup() throws Exception {
...
 
-        assertThat(activity.getTitle().toString(), is("Unit Test Example"));
+	    assertThat(activity.getTitle().toString(), is("Unit Test Example"));
```
Comment (dalinaum) [inline examples/unitTestExample/src/test/java/io/realm/examples/unittesting/ExampleActivityTest.java]: Same above.

File: examples/unitTestExample/src/test/java/io/realm/examples/unittesting/ExampleActivityTest.java [hunk 5]
```diff
@@ -222,15 +223,13 @@ public void shouldBeAbleToAccessActivityAndVerifyRealmInteractions() {
...
      * of a problem with Powermock: https://github.com/jayway/powermock/issues/649
      */
-    @Ignore("FIXME: Some problems mocking OKHttp")
     @Test
     public void shouldBeAbleToVerifyTransactionCalls() {
 
         // Create activity
-        ActivityController<ExampleActivity> controller = Robolectric.buildActivity(ExampleActivity.class).setup();
-        ExampleActivity activity = controller.get();
+	    ExampleActivity activity = Robolectric.buildActivity(ExampleActivity.class).create().start().resume().visible().get();
```
Comment (dalinaum) [inline examples/unitTestExample/src/test/java/io/realm/examples/unittesting/ExampleActivityTest.java]: Same above.

File: examples/unitTestExample/src/test/java/io/realm/examples/unittesting/ExampleActivityTest.java [hunk 6]
```diff
@@ -222,15 +223,13 @@ public void shouldBeAbleToAccessActivityAndVerifyRealmInteractions() {
...
 
-        assertThat(activity.getTitle().toString(), is("Unit Test Example"));
+	    assertThat(activity.getTitle().toString(), is("Unit Test Example"));
```
Comment (dalinaum) [inline examples/unitTestExample/src/test/java/io/realm/examples/unittesting/ExampleActivityTest.java]: Saeme above.

File: examples/unitTestExample/src/test/java/io/realm/examples/unittesting/ExampleActivityTest.java [hunk 7]
```diff
@@ -248,7 +247,7 @@ public void shouldBeAbleToVerifyTransactionCalls() {
         verify(mockRealm, times(5)).executeTransaction(Mockito.any(Realm.Transaction.class));
 
         // Call the destroy method so we can verify that the .close() method was called (below)
-        controller.destroy();
+	    activity.onDestroy();
```
Comment (dalinaum) [inline examples/unitTestExample/src/test/java/io/realm/examples/unittesting/ExampleActivityTest.java]: Same above.

**Comment by beeender** [comment]
@abennsir before we merge your PR, would you please to sign our  Contributor License Agreement first? Thanks a lot!!

> Realm welcomes all contributions! The only requirement we have is that, like many other projects, we need to have a [Contributor License Agreement](https://en.wikipedia.org/wiki/Contributor_License_Agreement) (CLA) in place before we can accept any external code. Our own CLA is a modified version of the Apache Software Foundation’s CLA.

> [Please submit your CLA electronically using our Google form](https://docs.google.com/forms/d/1bVp-Wp5nmNFz9Nx-ngTmYBVWVdwTyKj4T0WtfVm0Ozs/viewform?fbzx=4154977190905366979) so we can accept your submissions. The GitHub username you file there will need to match that of your Pull Requests. If you have any questions or cannot file the CLA electronically, you can email <help@realm.io>.

**Comment by beeender** [comment]
thanks a lot for your contribution!!

**Pull request merged**